jsdependencies-sbt-plugin/src/main/scala/org/scalajs/jsdependencies/sbtplugin/JSDependenciesPlugin.scala

```diff
@@ -0,0 +1,364 @@
+            new ExactFilter(JSDependencyManifest.ManifestFileName),
+            collectJar = jsDependencyManifestsInJar(_),
+            collectFile = { (file, _) =>
+              fromJSON[JSDependencyManifest](readJSON(IO.read(file)))
+            })
+
+        rawManifests.map(manifests => filter(manifests.toTraversable))
+      },
+
+      scalaJSNativeLibraries := {
+        collectFromClasspath(fullClasspath.value,
+            "*.js", collectJar = _.jsFiles,
+            collectFile = FileVirtualJSFile.relative)
+      },
+
+      resolvedJSDependencies := {
+        val dependencyFilter = jsDependencyFilter.value
+        val attLibs = scalaJSNativeLibraries.value
+        val attManifests = jsDependencyManifests.value
+
+        // Collect originating files
+        val realFiles = {
+          attLibs.get(scalaJSSourceFiles).get ++
+          attManifests.get(scalaJSSourceFiles).get
+        }
+
+        // Collect available JS libraries
+        val availableLibs = {
+          val libs = mutable.Map.empty[String, VirtualJSFile]
+          for (lib <- attLibs.data)
+            libs.getOrElseUpdate(lib.relativePath, lib)
+          libs.toMap
+        }
+
+        // Actually resolve the dependencies
+        val resolved = DependencyResolver.resolveDependencies(
+            attManifests.data, availableLibs, dependencyFilter)
+
+        Attributed.blank[Seq[ResolvedJSDependency]](resolved)
+            .put(scalaJSSourceFiles, realFiles)
+      },
+
+      // Add the resolved JS dependencies to the list of JS files given to envs
+      jsExecutionFiles := {
+        val deps = resolvedJSDependencies.value.data
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
+        val libs = jsEnv.value match {
+          case _: org.scalajs.jsenv.nodejs.NodeJSEnv =>
+            val libCache = new VirtualFileMaterializer(false)
+
+            for (dep <- deps) yield {
+              dep.info.commonJSName.fold {
+                dep.lib
+              } { commonJSName =>
+                val fname = libCache.materialize(dep.lib).getAbsolutePath
+                new MemVirtualJSFile(s"require-$fname").withContent(
+                  s"""$commonJSName = require("${escapeJS(fname)}");"""
+                )
+              }
+            }
+
+          case _ =>
+            deps.map(_.lib)
+        }
+
+        libs ++ jsExecutionFiles.value
+      },
+
+      scalaJSRequestsDOM := {
+        requiresDOM.?.value.getOrElse(
+            jsDependencyManifests.value.data.exists(_.requiresDOM))
+      }
+  )
+
+  lazy val compileSettings = configSettings
+
+  lazy val testSettings = Def.settings(
+      configSettings,
+
+      moduleName in packageJSDependencies := moduleName.value + "-test"
```

Comment:
We definitely *do* want it. It's currently a bug. I was actually wondering whether to preserve the bug during the extraction and fix it later, once it is in its own repo; or to fix it during the extraction. WDYT?